Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -376,54 +386,47 @@ public class Chat {
     
     /// Loads messages for the specified pagination parameters and updates ``ChatState/messages``.
     ///
+    /// - Important: Loading messages for the pagination index 0 resets ``ChatState/messages``.
+    ///
     /// - Parameters:
     ///   - message: The parent message id which has replies.
     ///   - pagination: The pagination configuration which includes a limit and a cursor.
     ///
     /// - Throws: An error while communicating with the Stream API.
     /// - Returns: An array of messages for the pagination.
-    public func loadMessages(with pagination: MessagesPagination) async throws -> [ChatMessage] {
+    @discardableResult public func loadMessages(with pagination: MessagesPagination) async throws -> [ChatMessage] {
         try await channelUpdater.loadMessages(with: channelQuery, pagination: pagination)
     }
     
     // MARK: -
     
-    /// Loads messages for the first page and updates ``ChatState/messages``.
-    ///
-    /// - Note: Loading the first page resets the ``ChatState/messages``.
-    ///
-    /// - Throws: An error while communicating with the Stream API.
-    public func loadMessagesFirstPage() async throws {
```

Comment:
Question: What would be the alternative of this one? What did we decide on?

Reply:
`chat.loadMessages(with: .init(pageSize: n, offset: 0))`

I added a commend to docs in `loadMessages`: `/// - Important: Loading messages for the pagination index 0 resets ``ChatState/messages``.` This is what the first page was doing.

Reply:
Oh, not offset, `parameter: nil`

Reply:
Fixed it:
`/// - Important: Loading messages for the pagination with parameter nil resets ``ChatState/messages``. Example `chat.loadMessages(with: MessagesPagination(pageSize: 30, parameter: nil))`.`

Reply:
What about `get()`? Shouldn't get do this as well?

Reply:
Uh-oh, yeah, `get` is a better answer, I will change it.